### ISSUE TITLE
[feature-request] Allow implicit function application syntax for nested index expressions

### ISSUE THREAD

**Issue by munael** [evhub/coconut#492]
Currently

```py
# we can do
ys = list(map(.[0], xs))
# and
ys = list(map(.[0, 1], xs))
# but not
ys = list(map(.[0][1], xs))
```

This nesting is supported in the case of `attrgetter`, but not `itemgetter`. Instead of compiling to a closure (which we have to do by hand right now, anyway), it can instead compile to an expression of a new built-in function that basically does a 'nested-get' on `__getitem__`-able objects.

**Comment by evhub** [comment]
@narfanar As a simple workaround, you could do
```python
ys = list(map(.[0] ..> .[1], xs))
```
which is probably what `.[0][1]` will get compiled to if this gets implemented.

**Comment by narfanar** [comment]
> @narfanar As a simple workaround, you could do
> 
> ```python
> ys = list(map(.[0] ..> .[1], xs))
> ```
> 
> which is probably what `.[0][1]` will get compiled to if this gets implemented.

Doesn't that still allocate a closure? Why not an `itemgetter` that supports nesting?

So:

```py
.[x][y][z] #=> itemgetter(x, y, z)
# where `x` et al can be tuples, so `.[i, j, k]` falls out of this naturally.
```

**Comment by pavelbraginskiy** [comment]
If this is implemented I think `.[x].someattr[y].otherattr` or other arbitrary combination of indexing and attribute access should be allowed.

**Comment by narfanar** [comment]
@pavelbraginskiy For that case it'd involve several nested getters of different kinds. I'm personally okay with `-> _[x].someattr[y].otherattr` for that case.

**Comment by evhub** [comment]
This is now available in `coconut>=1.6.0` (see #544).
